### PR TITLE
Packet season creation CLI

### PR DESCRIPTION
**Pull request by JoelEager**
Still needs some cleanup, but hey - it works!

### CODE REVIEW

File: packet/models.py [hunk 1]
```diff
@@ -20,7 +20,7 @@ class Freshman(db.Model):
     __tablename__ = "freshman"
     rit_username = Column(String(10), primary_key=True)
     name = Column(String(64), nullable=False)
-    onfloor = Column(Boolean, nullable=False)
+    onfloor = Column(Boolean, nullable=False)       # TODO: I don't think we need this column but I'm leaving it for now
```
Comment (devinmatte) [inline packet/models.py]: Why might we not need this? We need to know which freshmen are on floor

File: wsgi.py [hunk 2]
```diff
@@ -0,0 +1 @@
+from app import app
```
Comment (devinmatte) [inline wsgi.py]: Why do we need this file?

File: packet/commands.py [hunk 3]
```diff
@@ -15,22 +18,78 @@ def create_secret():
...
+        raise e
+
+    print("All freshmen = {}".format(allFreshmen))
+    print("On floor freshmen = {}".format(onfloorFreshmen), end="\n\n")
 
+    if input("Data ready; create the season? (Y/n): ").lower() != "y":
+        return
+
+    # Go ahead and create all Freshman entries now so FreshSignatures can be created iteratively
+    freshmen = {}
```
Comment (csssuf) [inline packet/commands.py]: I might name this something more specific like `freshman_db_entries` to distinguish it from `allFreshmen`.

File: packet/models.py (hunk 1 above)
Comment (JoelEager) [reply packet/models.py]: Unless I'm thinking about it wrong we don't. The only thing we need to know is what the onfloor freshmen are for a given packet which is already covered by the `FreshSignature` table.

File: wsgi.py (hunk 2 above)
Comment (JoelEager) [reply wsgi.py]: The flask cli will automatically look for `wsgi.py` so if it exists we don't need to set the startup file via a environment variable. That makes using the cli a bit more convenient so I wanted to put it back in.

[the remark this answers is not in the document]
Comment (JoelEager) [reply wsgi.py]: I'll fix that; thanks.

File: packet/commands.py [hunk 4]
```diff
@@ -15,22 +18,78 @@ def create_secret():
...
 
-    db.session.commit()
+    print("Parsing freshmen csv...")
+    allFreshmen = {}
+    onfloorFreshmen = {}
+
+    try:
+        with open(freshmen_csv) as freshmenFile:
+            for row in freshmenFile:
+                row = row.strip().split(",")
```
[the remark this answers is not in the document]
Comment (JoelEager) [reply packet/commands.py]: I can't believe I haven't come across that module before. Thanks for pointing me to it.

File: packet/commands.py [hunk 5]
```diff
@@ -15,22 +18,78 @@ def create_secret():
...
+                    onfloorFreshmen[row[3]] = row[0]
+
+    except Exception as e:
+        print("Things broke")
+        raise e
+
+    print("All freshmen = {}".format(allFreshmen))
+    print("On floor freshmen = {}".format(onfloorFreshmen), end="\n\n")
 
+    if input("Data ready; create the season? (Y/n): ").lower() != "y":
```
[the remark this answers is not in the document]
Comment (JoelEager) [reply packet/commands.py]: Oh, that's what that means? Thanks for explaining that.

File: packet/commands.py [hunk 6]
```diff
@@ -15,22 +19,108 @@ def create_secret():
...
+        else:
+            # This freshman is already in the DB so just update them
+            freshmen_in_db[csv_freshman.rit_username].onfloor = csv_freshman.onfloor
+            freshmen_in_db[csv_freshman.rit_username].name = csv_freshman.name
 
-    packet = Packet(freshman=freshman)
-    db.session.add(packet)
+    # Update all freshmen entries that represent people who are no longer freshmen
+    for freshman in filter(lambda freshman: freshman.rit_username not in freshmen_in_csv, freshmen_in_db.values()):
+        freshman.onfloor = False
```
[the remark this answers is not in the document]
Comment (devinmatte) [reply packet/commands.py]: There is also the case where if the layout of floor changes mid packet, we want to be able to rebuild packets with the data

**Pull request merged**